snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule671.groovy

```diff
@@ -0,0 +1,74 @@
+package scripts;
+
+import com.b2international.index.Hits
+import com.b2international.index.query.Expressions
+import com.b2international.index.query.Query
+import com.b2international.index.query.Expressions.ExpressionBuilder
+import com.b2international.index.revision.RevisionSearcher
+import com.b2international.snowowl.core.ComponentIdentifier
+import com.b2international.snowowl.core.date.EffectiveTimes
+import com.b2international.snowowl.snomed.common.SnomedRf2Headers
+import com.b2international.snowowl.snomed.common.SnomedTerminologyComponentConstants
+import com.b2international.snowowl.snomed.common.SnomedConstants.Concepts
+import com.b2international.snowowl.snomed.core.domain.SnomedDescription
+import com.b2international.snowowl.snomed.core.domain.SnomedDescriptions
+import com.b2international.snowowl.snomed.datastore.index.entry.SnomedConceptDocument
+import com.b2international.snowowl.snomed.datastore.index.entry.SnomedRefSetMemberIndexEntry
+import com.b2international.snowowl.snomed.datastore.request.SnomedRequests
+import com.google.common.collect.Lists
+import com.google.common.collect.Maps
+import com.google.common.collect.Sets
+
+final Set<ComponentIdentifier> issues = Sets.newHashSet()
+final RevisionSearcher searcher = ctx.service(RevisionSearcher.class)
+
+ExpressionBuilder filterExpressionBuilder = Expressions.builder()
+		.filter(SnomedConceptDocument.Expressions.inactive())
+
+if (params.isUnpublishedOnly) {
+	filterExpressionBuilder.filter(SnomedConceptDocument.Expressions.effectiveTime(EffectiveTimes.UNSET_EFFECTIVE_TIME))
+}
+
+List<String> inactiveIds = Lists.newArrayList()
```

Comment:
Use `Set` for distinct values and when you don't actually need the index.

Reply:
Done